src/Stratis.Bitcoin.Features.Api/Startup.cs

```diff
@@ -69,32 +71,37 @@ public void ConfigureServices(IServiceCollection services)
                 .AddJsonOptions(options => Utilities.JsonConverters.Serializer.RegisterFrontConverters(options.SerializerSettings))
                 .AddControllers(this.fullNode.Services.Features, services);
 
-            // Register the Swagger generator, defining one or more Swagger documents
-            services.AddSwaggerGen(setup =>
+            // Enable API versioning.
+            // Note much of this is borrowed from https://github.com/microsoft/aspnet-api-versioning/blob/master/samples/aspnetcore/SwaggerSample/Startup.cs
+            services.AddApiVersioning(options =>
             {
-                setup.SwaggerDoc("v1", new Info { Title = "Stratis.Bitcoin.Api", Version = "v1" });
+                // Our versions are configured to be set via URL path, no need to read from querystring etc.
+                options.ApiVersionReader = new UrlSegmentApiVersionReader();
 
-                //Set the comments path for the swagger json and ui.
-                string basePath = PlatformServices.Default.Application.ApplicationBasePath;
-                string apiXmlPath = Path.Combine(basePath, "Stratis.Bitcoin.Api.xml");
-                string walletXmlPath = Path.Combine(basePath, "Stratis.Bitcoin.LightWallet.xml");
```

Comment:
Ah I see it was copied from here... Still, do you what the significance is?

Reply:
I have no idea. Just copied it over. My guess is at some point previously there was documentation missing from some of the APIs? 

We could experiment with removing it, but probably best for a different PR.

Reply:
from what I know swagger can read up generated xml doc to fill up its API description using comment, I don't know if we are using it properly or not (and of course the build have to generate these xml)